src/basic/user-util.c

```diff
@@ -668,6 +668,15 @@ bool valid_home(const char *p) {
         return true;
 }
 
+bool valid_shell(const char *p) {
```

Comment:
Should it be inline and moved to the header file?

Reply:
Yeah... I guess. It's inline now.